### PR TITLE
IPC bugsquashing and minor fixes.

### LINKED ISSUE #519: IPCS can take cellular damage
## Round ID: 
7697 beestation lrp

## Reproduction:
Get a slime latched onto you. they do some cellular damage
(me, the ipc mime)
me!

bruh!

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/81533361-ipcs-can-take-cellular-damage?utm_campaign=plugin&utm_content=tracker%2F120077027&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F120077027&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### LINKED ISSUE #550: Bugged masks
Issue reported from  Round ID: 7971 (BeeStation Golden - For Beginners! Low RP Chaotic Fun!)

Reporting client version: 512.1470

I think there is something bugged with masks, you can eat while mask still covers your mouth, there is a message 'you have to remove your mask first' but that's it, still eating food. I'm talking about breath masks, but as far as I recall it works for gas ones too, but not 100% sure about that.

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/81987882-bugged-masks?utm_campaign=plugin&utm_content=tracker%2F120077027&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F120077027&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### LINKED ISSUE #583: IPCs Get Suffocation damage when revived
You get suffocation damage when getting revived as an IPC, Which makes you sleep forever.
The only way to wake up is with an admin healing you.

### PR DESCRIPTION
**Pull request by SomeAngryMiner**
## About The Pull Request

This Fixes a handful of problems related to IPC, and some minor, easily fixable ones.

-IPC no longer take oxyloss on critical state, should be only the overheat damage, now.

-Liquid solder now cures brain traumas that dont require lobotomy or surgery. Mirror from oracle.

-Slimes can no longer latch onto IPC, but they still can attack them without any repercutions.

-Vanilla Icecream is no longer /Datum/reagent/consumable/vanilla, as tasty as that sounds.

-Fixes an oversight of mine on gasmasks, they now properly block you from eating while on.

-Removes duplicated surgery message while handling IPC posibrains on surgery, wich now you can reattach, cheers.

In resume, fixes #583 , fixes #550 and fixes #519 , amongst other things.

## Why It's Good For The Game

fix good k

## Changelog
:cl:
fix: IPC no longer take oxyloss on crit
fix: no longer eating through gasmask
fix: Fixes minor bugs on IPC surgery.
fix: Vanilla icrecream is no longer a visible datum, its plain, normal, icreacream now.
tweak: liquid solder now cures brain traumas that dont require surgery on IPC
balance: Slimes can no longer latch on IPC, but can still attack them.
/:cl:

### CODE REVIEW

**Comment by SomeAngryMiner** [comment]
there we go.

File: code/modules/mob/living/simple_animal/slime/powers.dm [hunk 1]
```diff
@@ -72,6 +72,12 @@
 		to_chat(src, "<span class='warning'><i>I can't latch onto another slime...</i></span>")
 		return FALSE
 
+	if(isIPC)
```
Comment (MarkSuckerberg) [inline code/modules/mob/living/simple_animal/slime/powers.dm]: For whatever reason, this is throwing an error. Apparently `isIPC` isn't defined for this instance

**Comment by despong** [comment]
synthahol??? maybe???

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm [hunk 2]
```diff
@@ -423,6 +423,12 @@
 
 /datum/reagent/medicine/liquid_solder/on_mob_life(mob/living/M)
 	M.adjustBrainLoss(-3*REM)
+	if(iscarbon(M))
```
Comment (MarkSuckerberg) [inline code/modules/reagents/chemistry/reagents/medicine_reagents.dm:426]: Won't this allow all carbons to use it? Or is it supposed to be like that

**Comment by SomeAngryMiner** [comment]
i thought of making another chemical, but wouldnt make sense to have yet another one just for a single species to use, perhaps ill add a synth version of psicodine at most, someday.

**Comment by MarkSuckerberg** [comment]
Ah, alright

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm (hunk 2 above)
Comment (SomeAngryMiner) [reply code/modules/reagents/chemistry/reagents/medicine_reagents.dm:426]: it checks for the reagent processing tag first.

Comment (MarkSuckerberg) [reply code/modules/reagents/chemistry/reagents/medicine_reagents.dm:426]: But, after that, there aren't any IPC checks as far as I can see.

Comment (SomeAngryMiner) [reply code/modules/reagents/chemistry/reagents/medicine_reagents.dm:426]: if the person isnt a synthetic, it wont process the reagent at all.

Comment (MarkSuckerberg) [reply code/modules/reagents/chemistry/reagents/medicine_reagents.dm:426]: Oh, didn't see the `process_flags = SYNTHETIC` above. Apologies

Comment (SomeAngryMiner) [reply code/modules/reagents/chemistry/reagents/medicine_reagents.dm:426]: its okay bro.

File: code/modules/surgery/organ_manipulation.dm [hunk 3]
```diff
@@ -95,7 +95,6 @@
 			to_chat(user, "<span class='notice'>There is no room for [I] in [target]'s [parse_zone(target_zone)]!</span>")
 			return -1
 	if(istype(tool, /obj/item/organ/brain/positron))
-		current_type = "posibrain"
```
Comment (qwertyquerty) [inline code/modules/surgery/organ_manipulation.dm:98]: Why was this removed?

File: code/modules/reagents/reagent_containers.dm [hunk 4]
```diff
@@ -12,6 +12,7 @@
 	var/spawned_disease = null
 	var/disease_amount = 20
 	var/spillable = FALSE
+	var/you_drink_from_this = FALSE
```
Comment (qwertyquerty) [inline code/modules/reagents/reagent_containers.dm]: Bad variable name

File: code/modules/surgery/organ_manipulation.dm (hunk 3 above)
Comment (SomeAngryMiner) [reply code/modules/surgery/organ_manipulation.dm:98]: it conflicted with the surgery, somehow, if i didnt remove it you couldnt place the brain back.

File: code/modules/reagents/reagent_containers.dm (hunk 4 above)
Comment (SomeAngryMiner) [reply code/modules/reagents/reagent_containers.dm]: blame flattest for the original pr, ill change it.

Comment (SomeAngryMiner) [reply code/modules/reagents/reagent_containers.dm]: done.

**Comment by despong** [comment]
change x2 burn to x1.7 and brute to x1.4 because i literally cannot stress how fucking bullshit an instant crit from shocked doors is

edit : how do they even take that much damage is beyond me since metal is a really fast heat conductor

**Comment by SomeAngryMiner** [comment]
i can perfectly assure you that x2 burn isnt an instant crit of interacting with shocked doors.
Balancing IPC damage values would require a poll on the community, wich will probably do later.

**Comment by despong** [comment]
bruh

**Comment by qwertyquerty** [comment]
The IPC and Abductor drinking stuff does not appear to be the most quality code in the world. What do you think @ike709

**Comment by ike709** [comment]
@SomeAngryMiner Instead of commenting on every issue you think this will fix, put `Fixes #whatever` in the PR description. Won't merge this until you do.

**Comment by ike709** [comment]
@qwertyquerty I agree.

I think making it a trait would be a little excessive, but also the best non-snowflake way to do it.

**Comment by SomeAngryMiner** [comment]
the descriptions where there, but i added the fix tags, forgot about those for a second.

**Comment by qwertyquerty** [comment]
Undo the IPC and Abductor drinking stuff and I will merge. The code for it isn't the best and I really don't see a need for them to drink.

**Comment by SomeAngryMiner** [comment]
Fair enough, the code was so IPCs would be able to consume the required chemicals for maintenance without the need of a piercing syringe, Abductors just got it as rebound by having NOMOUTH trait.

**Comment by SomeAngryMiner** [comment]
aaaand its done.
